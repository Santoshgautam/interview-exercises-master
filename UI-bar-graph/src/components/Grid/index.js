import React from "react";

const Grid = ({ maxStep, step, start }) => {
  const grids = [];
  let value = maxStep;

  while (value >= start) {
    grids.push(
      <div className="grid-y" key={value}>
        <div className="grid-label">{value}</div>
        <div className="grid-line"></div>
      </div>
    );
    value -= step;
  }
  return (
    <div className="axis-y">
      <div className="axis"></div>
      {grids}
    </div>
  );
};

export default Grid;
