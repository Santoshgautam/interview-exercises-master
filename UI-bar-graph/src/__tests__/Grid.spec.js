import Grid from "../components/Grid";

describe("Grid Component", () => {
  let wrapper, consoleError;

  beforeEach(() => {
    wrapper = shallow(<Grid />);
    consoleError = console.error;
    console.error = jest.spyOn(console, "error");
  });

  afterEach(() => {
    console.error = consoleError;
  });

  it("renders without crashing", () => {
    expect(wrapper.exists()).toBe(true);
  });

  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("displays correct number of grid lines", () => {
    const sample = require("../constants/data.json");
    const step = 10;
    const maxStep = Math.ceil(sample.data.reduce((r, d) => (r < d ? d : r), 0) / 10) * 10;

    wrapper = mount(<Grid maxStep={maxStep} start={0} step={step} />);
    expect(wrapper.find(".grid-line")).toHaveLength(maxStep / step + 1);
  });
});
