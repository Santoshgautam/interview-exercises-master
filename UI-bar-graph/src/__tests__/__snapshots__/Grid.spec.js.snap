// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Component matches the snapshot 1`] = `ShallowWrapper {}`;
